### PR TITLE
Create specific parser implementations

### PR DESCRIPTION
**Pull request by MartinYSpasov**

### CODE REVIEW

File: event-sourcing/event-subscription/src/main/java/uk/gov/justice/subscription/ParserProducer.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@ApplicationScoped
+public class ParserProducer {
+
+    private YamlFileValidator yamlFileValidator;
```
Comment (mapingo) [inline event-sourcing/event-subscription/src/main/java/uk/gov/justice/subscription/ParserProducer.java]: yamlFileValidator never gets assigned in the code below

File: event-sourcing/subscription-manager/pom.xml [hunk 2]
```diff
@@ -48,6 +48,27 @@
...
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-subscription-test-utils</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.framework-api</groupId>
+            <artifactId>framework-api-subscription</artifactId>
+            <version>${framework-api.version}</version>
+        </dependency>
```
Comment (mapingo) [inline event-sourcing/subscription-manager/pom.xml]: This dependency needs to be moved above the test dependency section

File: generators/messaging-adapter-generator/src/main/java/uk/gov/justice/subscription/jms/parser/SubscriptionWrapper.java [hunk 3]
```diff
@@ -13,9 +13,9 @@
     private final Map<String, EventSource> eventSourceMap;
     private final SubscriptionDescriptor subscriptionDescriptor;
 
-    public SubscriptionWrapper(final SubscriptionDescriptor subscriptionDescriptor, final List<EventSource> eventSources) {
+    public SubscriptionWrapper(final SubscriptionDescriptor subscriptionDescriptor, final List<EventSource> EventSources) {
```
Comment (mapingo) [inline generators/messaging-adapter-generator/src/main/java/uk/gov/justice/subscription/jms/parser/SubscriptionWrapper.java]: EventSources needs to start with a lowercase 'e'

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/core/SubscriptionJmsEndpointGeneratorTest.java [hunk 4]
```diff
@@ -592,14 +592,14 @@ public void shouldOnlyCreateMessageSelectorForPostActionAndIgnoreAllOtherActions
...
-       final String jmsUri = "jms:topic:people.controller.command";
+        final String jmsUri = "jms:topic:people.controller.command";
 
         final Event event = event()
                 .withName("people.command1")
                 .withSchemaUri("http://justice.gov.uk/json/schemas/domains/example/people.command1.json")
                 .build();
 
-        Event event2  = event()
+        Event event2 = event()
```
Comment (mapingo) [inline generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/core/SubscriptionJmsEndpointGeneratorTest.java]: can be final

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTest.java [hunk 5]
```diff
@@ -88,7 +58,7 @@ public void shouldThrowExceptionIfNoEventSourcesPresent() {
         try {
             eventSourcesFileParser.getEventSources(baseDir, emptyList());
             fail();
-        } catch (final SubscriptionFileParserException e) {
+        } catch (final Exception e) {
```
Comment (mapingo) [inline generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTest.java]: No assertion that the correct Exception is thrown

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTestDefinition.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Test
+    public void shouldThrowExceptionIfNoEventSourcesPresent() {
+        final Path baseDir = mock(Path.class);
+
+        try {
+            eventSourcesFileParser.getEventSources(baseDir, emptyList());
+            fail();
+        } catch (final Exception e) {
+            assertThat(e.getMessage(), is("No event-sources.yaml files found!"));
```
Comment (mapingo) [inline generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTestDefinition.java]: No assertion that the correct Exception is thorwn

File: event-sourcing/event-subscription-registry/src/main/java/uk/gov/justice/subscription/registry/EventSourceRegistryProducer.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.stream.Stream;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+public class EventSourceRegistryProducer {
+
+    @Inject
+    private EventSourcesParser eventSourcesParser;
```
[the remark this answers is not in the document]
Comment (MartinYSpasov) [reply event-sourcing/event-subscription-registry/src/main/java/uk/gov/justice/subscription/registry/EventSourceRegistryProducer.java]: changed

File: event-sourcing/event-subscription-registry/src/main/java/uk/gov/justice/subscription/registry/EventSourceRegistryProducer.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Inject
+    private YamlFileFinder yamlFileFinder;
```
[the remark this answers is not in the document]
Comment (MartinYSpasov) [reply event-sourcing/event-subscription-registry/src/main/java/uk/gov/justice/subscription/registry/EventSourceRegistryProducer.java]: changed

File: event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryProducerTest.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubscriptionDescriptorRegistryProducerTest {
+
+    @Mock
+    YamlFileFinder yamlFileFinder;
```
[the remark this answers is not in the document]
Comment (MartinYSpasov) [reply event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryProducerTest.java]: done

File: event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryProducerTest.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Mock
+    SubscriptionDescriptorsParser subscriptionDescriptorsParser;
```
[the remark this answers is not in the document]
Comment (MartinYSpasov) [reply event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryProducerTest.java]: done

File: event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryTest.java [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+        try {
+            subscriptionDescriptorRegistry.getSubscription(thisSubscriptionDoesNotExist);
+            fail();
+        } catch (final RegistryException expected) {
+            assertThat(expected.getMessage(), is("Failed to find subscription 'thisSubscriptionDoesNotExist' in registry"));
+        }
+    }
+
+    @Test
+    public void shouldThrowASubscriptionManagerProducerExceptionIfNoSubscriptionCanBeFoundWithTheSpecifiedSubscriptionName1() throws Exception {
```
[the remark this answers is not in the document]
Comment (MartinYSpasov) [reply event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/SubscriptionDescriptorRegistryTest.java]: done

File: event-sourcing/subscription-manager/pom.xml (hunk 2 above)
Comment (MartinYSpasov) [reply event-sourcing/subscription-manager/pom.xml]: done

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/core/SubscriptionJmsEndpointGeneratorTest.java (hunk 4 above)
Comment (MartinYSpasov) [reply generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/core/SubscriptionJmsEndpointGeneratorTest.java]: done

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTest.java (hunk 5 above)
Comment (MartinYSpasov) [reply generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTest.java]: done

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTestDefinition.java (hunk 6 above)
Comment (MartinYSpasov) [reply generators/messaging-adapter-generator/src/test/java/uk/gov/justice/subscription/jms/parser/EventSourcesFileParserTestDefinition.java]: done

**Pull request merged**